lib/sles4sap/sap_deployment_automation_framework/deployment_connector.pm

```diff
@@ -209,15 +209,36 @@ Deployment ID returned from both jobs: 123456 - because it matches with existing
 sub find_deployment_id {
     my (%args) = @_;
     return get_var('SDAF_DEPLOYMENT_ID') if get_var('SDAF_DEPLOYMENT_ID');
+    return get_var('SDAF_GRANDPARENT_ID') if get_var('SDAF_GRANDPARENT_ID');
+
     $args{deployer_resource_group} //= get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP');
     my @check_list = (get_current_job_id(), @{get_parent_ids()});
 
+    # For SDAF cleanup job it needs its grandparent's id
+    my @parents = get_parent_ids();
+    my $parent_id = $parents[0][0];
+    if (looks_like_number($parent_id) && @parents == 1) {
+        # Get the SDAF deployment job ID, here it is set via SDAF_GRANDPARENT_ID and saved in vars.json
+        my $vars_json = 'vars.json';
+        my $openqa_url = get_var('OPENQA_URL', 'openqa.suse.de');
+        script_run("wget --no-check-certificate -O /tmp/$vars_json https://$openqa_url/tests/$parent_id/file/$vars_json");
+        my $str = script_output("grep SDAF_GRANDPARENT_ID /tmp/$vars_json | cut -d ':' -f2", proceed_on_failure => 1);
+        if ($str) {
+            my ($id) = $str =~ /(\d+)/;
+            @check_list = ($id) if ($id);
```

Comment:
this results in the original values in the array will be covered, right? is this expected?

Reply:
> this results in the original values in the array will be covered, right? is this expected?

Yes.